Car.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <memory>
+#include "InvalidsError.hpp"
+#include "PetrolEngine.hpp"
+
+class Car {
+public:
+    virtual ~Car();
+    void turnLeft();
+    void turnRight();
+    void brake();
+    void accelerate(int speed);
+    void showSpeed();
+    int getSpeed();
+    virtual void restore() = 0;
+    virtual void getInfoEngine() = 0;

```

Comment:
Might be marked as const method.
And maybe "print" would be better than "get" - it's strange that "get" method does not return anything.